packages/p2p/src/stores/order-store.js

```diff
@@ -471,33 +483,44 @@ export default class OrderStore {
                 verification_code,
                 dry_run: 1,
             }).then(response => {
-                this.root_store.general_store.hideModal();
+                general_store.hideModal();
                 if (response) {
                     if (!response.error) {
                         clearTimeout(wait);
                         const wait = setTimeout(
-                            () => this.root_store.general_store.showModal({ key: 'EmailLinkVerifiedModal', props: {} }),
+                            () => general_store.showModal({ key: 'EmailLinkVerifiedModal', props: {} }),
                             650
                         );
-                    } else if (
-                        response.error.code === api_error_codes.INVALID_VERIFICATION_TOKEN ||
-                        response.error.code === api_error_codes.EXCESSIVE_VERIFICATION_REQUESTS
-                    ) {
-                        clearTimeout(wait);
-                        this.setVerificationLinkErrorMessage(response.error.message);
-                        const wait = setTimeout(() => {
-                            this.root_store.general_store.showModal({
-                                key: 'InvalidVerificationLinkModal',
-                                props: { error_message: response.error.message, order_id },
+                    } else if (response.error) {
+                        const { code, message } = response?.error;
+
+                        if (
+                            code === api_error_codes.INVALID_VERIFICATION_TOKEN ||
+                            code === api_error_codes.EXCESSIVE_VERIFICATION_REQUESTS
+                        ) {
+                            clearTimeout(wait);
+                            this.setVerificationLinkErrorMessage(message);
+                            const wait = setTimeout(() => {
```

Comment:
@farrah-deriv do we still need this `setTimeout`? If its related to modal transition glitch, it should already be handled by the modal manager

Reply:
@adrienne-deriv yes, we still need it since we are using the CSS transition of modal component.